core/src/main/java/org/elasticsearch/common/settings/Setting.java

```diff
@@ -96,6 +96,11 @@
         Dynamic,
 
         /**
+         * mark this setting as final (the value cannot be updated)
```

Comment:
I think it is worth explaining here how this is different than not marking the setting as `Dynamic`.